Source/DafnyLanguageServer.Test/Unit/DafnyLangSymbolResolverTest.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using Microsoft.Boogie;
+using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.Dafny.LanguageServer.Language.Symbols;
+using Microsoft.Dafny.LanguageServer.Workspace;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Unit; 
+
+[TestClass]
+public class DafnyLangSymbolResolverTest {
+  private DafnyLangSymbolResolver dafnyLangSymbolResolver;
+
+  [TestInitialize]
+  public void SetUp() {
+    var loggerFactory = new Mock<ILoggerFactory>();
+    dafnyLangSymbolResolver = new DafnyLangSymbolResolver(
+      loggerFactory.Object.CreateLogger<DafnyLangSymbolResolver>()
+    );
+  }
+
+  class CollectingErrorReporter : BatchErrorReporter {
+    public Dictionary<ErrorLevel, List<ErrorMessage>> GetErrors() {
+      return this.AllMessages;
+    }
+  }
+
+  class DummyModuleDecl : LiteralModuleDecl {
+    public DummyModuleDecl() : base(
+      new DefaultModuleDecl(), null) {
+    }
+    public override object Dereference() {
+      return this;
+    }
+  }
+
+  [TestMethod]
+  public void EnsureResilienceAgainstErrors() {
+    // Builtins is null to trigger an error.
```

Comment:
It seems like this test relies on some badly defined API of `DafnyPipeline.csproj` to trigger the exception. I think it might be nicer if the test uses a more explicit API to trigger a failure, like mocking something from `DafnyPipeline.csproj` and letting the mock throw an Exception, but I imagine `DafnyPipeline.csproj` is not setup for being mocked.

Reply:
Yes you identified this test correctly.
I don't see either how to mock the DafnyPipeline, so I took the route of the minimum working test for now.